.github/workflows/dep_check.yml

```diff
@@ -0,0 +1,18 @@
+name: DepCheck
```

Comment:
I'm wondering if we should have these new workflows as new jobs of our CI workflow rather than creating new workflows.

Reply:
also I'm wondering if it makes sense to implement depcheck and semgrep as tox actions so that it's easier to run such linters locally as part of the tox -e code-linters

Reply:
> I'm wondering if we should have these new workflows as new jobs of our CI workflow rather than creating new workflows.

Is there a downside to using new workflows?


> also I'm wondering if it makes sense to implement depcheck and semgrep as tox actions so that it's easier to run such linters locally as part of the tox -e code-linters

I opted not to do this because for [dependency check specifically](https://github.com/jeremylong/DependencyCheck) the installation is platform-specific.